### DISCUSSION TITLE
Having trouble creating and seeding a torrent. Download machine never discovers the seed machine.

### DISCUSSION THREAD

**Discussion by Erotemic** [transmission/transmission#6962] [Q&A]
I'm attempting to create a torrent for custom data and I would like to track that torrent with the Mainline DHT. This is in an effort to compare the utility of bittorrent and IPFS, so I explicitly don't want to use a specific tracker. I'm looking to independently validate some of the claims in this document: https://gist.github.com/liamzebedee/4be7d3a551c6cddb24a279c4621db74c

Before I get into details, I want to preface this by saying: so far all my testing is on a LAN behind a standard consumer-level router, so I'm wondering if that would cause issues with relying on the DHT. If that's the issue, is there a way I can fix that?

If not, then here is my setup, I have two Linux machines on a LAN. At the moment, I've disabled the ufw firewall on each of them to rule that out as the issue. On each machine I've installed transmission: 

```bash
# Install Transmission
sudo apt-get install transmission-cli

# Install the GUI as well, although this is not needed.
sudo apt install transmission-gtk
```

And configured transmission to enable local-peer-discovery:

```bash
# Enable local peer discovery in the settings
sed -i 's/"lpd-enabled": false/"lpd-enabled": true/' ~/.config/transmission/settings.json
# Verify it is enabled
cat ~/.config/transmission/settings.json | grep lpd
```

On the seed machine I create some demo data, create a torrent, and then I start seeding it: 

```bash
# Create a dummy set of data that will be shared
WORKING_DPATH=$HOME/tmp/create-torrent-demo
DATA_DPATH=$WORKING_DPATH/shared-demo-data-v1
TORRENT_FPATH=$WORKING_DPATH/shared-demo-data-v1.torrent

mkdir -p "$WORKING_DPATH"
cd $WORKING_DPATH

mkdir -p "$DATA_DPATH"
echo "some data" > $DATA_DPATH/data1.txt
echo "some data" > $DATA_DPATH/data2.txt
echo "some other data" > $DATA_DPATH/data3.txt

transmission-create --comment "a demo torrent v1" --outfile "$TORRENT_FPATH" "$DATA_DPATH"

cat "$TORRENT_FPATH"

# Start seeding the torrent
# Ensure that the download directory contains the data to be seeded
transmission-cli --verify --download-dir "$(dirname $DATA_DPATH)" $TORRENT_FPATH
```

Then on the downloading machine, I grab the torrent and attempt to download it: 

```bash
# Transfer the torrent to the download machine
SEEDING_MACHINE_NAME=remote
rsync $SEEDING_MACHINE_NAME:tmp/create-torrent-demo/shared-demo-data-v1.torrent .

# Add the torrent to transmission
transmission-cli shared-demo-data-v1.torrent
```

For some reason, the torrents are paused when I add them via `transmission-cli`, so on each machine I just fire up `transmission-gtk` and manually click start, although I suspect there must be a better way. Ideally I understand this procedure well enough that I can  use it entirely from the command line. 

In any case, at this point I would expect to start seeing some data being transferred, but both machines seem to be unaware of each other wrt to this torrent. In the transmission gui the seeding machine says: "Seeding to 0 of 0 connected peers" and the downloading machine says "Downloading from 0 of 0 peers". 

So for some reason the download machine is unable to fine the seeding machine. I'm not sure why this is. Local peer discovery should be on in both cases. Did I miss a step? From what I understand all non-private torrents are announced to the DHT. Is it simply a case of waiting for it to be resolved? Any help is appreciated.

**Comment by philrhc** [comment]
hmm.. i'm trying something similar to this, two transmissions running on different ports can see discover each other and i can see LDP working, the torrent successfully downloads on the second instance, it took a while for the discovery to happen...
changing message_level to 5 (debug) might help, 6 (trace) can be very noisy
also just making sure that you're restarting after changing the settings.json ?
good luck!

**Reply by Erotemic** [reply]
I posted some more information on super user (https://superuser.com/questions/1851016/a-minimal-example-of-sharing-data-with-bittorrent?noredirect=1#comment2933483_1851016) but they didn't seem to like the question. 

I've backed up what I wrote here: https://gist.github.com/Erotemic/6e5941582fd3d95e04d133ae6b401fea as I think they may delete it.

I did find a resolution:

> To anyone in the future who finds this: the answer was permissions in the download directory (on the download machine) and permission on the data and torrent in the seed machine. When I switched to doing everything in /var/lib/transmission/downloads and ensured the transmission-daemon user could read/write all files to to all dirs, it worked. –

I will also try changing the message level when I work on debugging LDP more. Thanks!
